framework/auth/core.py

```diff
@@ -42,17 +43,19 @@
 logger = logging.getLogger(__name__)
 
 
-# Hide implementation of token generation
-def generate_confirm_token():
-    return security.random_string(30)
-
-
-def generate_claim_token():
+# verification key v1
+def generate_verification_key():
     return security.random_string(30)
 
 
-def generate_verification_key():
-    return security.random_string(30)
+# verification key v2, expires in 30 minutes
+def generate_verification_key_v2():
```

Comment:
This works for forgot/reset password verification and contributor-ship claims.  But for send/resend confirmation, `token` is used as key of the map. I will leave this unchanged.

```
User.email_verifications:
{
    <token> : {
        'email': <email address>,
        'expiration': <datetime>,
        'confirmed': whether user is confirmed or not
    }
}
```